subprojects/core/src/test/groovy/org/gradle/api/internal/DefaultTaskTest.groovy

```diff
@@ -59,7 +59,7 @@ class DefaultTaskTest extends AbstractTaskTest {
 
     def "default task"() {
         given:
-        def identity = TaskIdentity.create(TEST_TASK_NAME, Task, project)
+        def identity = TaskIdentity.create(TEST_TASK_NAME, Task, project, 1)
```

Comment:
🤔 I think these should use the `TaskIdentityFactory`. We could add a test fixture so it is easy to create the factory, similarly to `TestFiles`.

Reply:
I created a `TestTaskIdentities` fixture that provides a factory with a dummy `ConfigurationCacheableIdFactory` delegate, and I refactored all the tests to use it instead of directly accessing the `TaskIdentity.create`.

Probably best to look at the last commit in isolation:
https://github.com/gradle/gradle/pull/24375/commits/25c903b1fee469130086c88dac144fdf469e1668